File: mycollab-ui/src/main/java/com/mycollab/vaadin/spring/VaadinSpringConfiguration.java
package com.mycollab.vaadin.spring;

import com.mycollab.vaadin.MyCollabServlet;
import org.springframework.boot.web.servlet.ServletRegistrationBean;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
@Configuration
@Profile("production")
public class VaadinSpringConfiguration {
    @Bean
    public ServletRegistrationBean mainServlet() {
        ServletRegistrationBean bean = new ServletRegistrationBean(
                new MyCollabServlet(), "/*");
        bean.setLoadOnStartup(1);
        return bean;
    }
}

File: mycollab-web/src/main/java/com/mycollab/web/configuration/VaadinSpringConfiguration.kt
package com.mycollab.web.configuration

import com.mycollab.web.AppServlet
import com.vaadin.spring.annotation.EnableVaadin
import org.springframework.boot.web.servlet.ServletRegistrationBean
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.vaadin.spring.events.annotation.EnableEventBus

@Configuration
@EnableVaadin
@EnableEventBus
class VaadinSpringConfiguration {

    @Bean("mainServlet")
    fun mainServlet(): ServletRegistrationBean<*> {
        val bean = ServletRegistrationBean(AppServlet(), "/*")
        bean.setLoadOnStartup(1)
        return bean
    }
}
